OBAKit/UI/MVVM/OBAStaticTableViewController.m

```diff
@@ -334,14 +336,31 @@ - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath
     }
 }
 
-- (BOOL)tableView:(UITableView *)tableView canEditRowAtIndexPath:(NSIndexPath *)indexPath {
-    OBABaseRow *row = [self rowAtIndexPath:indexPath];
-    return row.rowActions.count > 0;
-}
-
-- (NSArray<UITableViewRowAction *> *)tableView:(UITableView *)tableView editActionsForRowAtIndexPath:(NSIndexPath *)indexPath {
+- (UISwipeActionsConfiguration *)tableView:(UITableView *)tableView trailingSwipeActionsConfigurationForRowAtIndexPath:(NSIndexPath *)indexPath {
     OBABaseRow *row = [self rowAtIndexPath:indexPath];
-    return row.rowActions;
+    UISwipeActionsConfiguration *actionsConfiguration = [row.rowActions copy];
+    
+    // Add a delete action if a delete model exists.
+    if (row.deleteModel) {
+        UISwipeActionsConfiguration *configurationWithDeleteAction;
+        
+        UIContextualAction *deleteAction = [UIContextualAction contextualActionWithStyle:UIContextualActionStyleDestructive title:OBAStrings.delete handler:^(UIContextualAction *action, UIView *sourceView, void (^completionHandler)(BOOL)) {
+            
+            BOOL deleteStatus = [self deleteRowAtIndexPath: indexPath];
+            completionHandler(deleteStatus);
+        }];
+        
+        if (!actionsConfiguration) {
+            configurationWithDeleteAction = [UISwipeActionsConfiguration configurationWithActions: @[deleteAction]];
+        } else {
+            NSArray<UIContextualAction*> *actions = [[actionsConfiguration actions] arrayByAddingObject:deleteAction];
+            configurationWithDeleteAction = [UISwipeActionsConfiguration configurationWithActions:actions];
+        }
+        
+        return configurationWithDeleteAction;
+    } else {
+        return row.rowActions;
```

Comment:
In contrast with what I wrote above, I prefer early returns over multiple layers of `if` statement nesting, even when that requires a negation. That way, instead of two levels of `if` statement nesting in this method, you can have a few single layers with an early return at the beginning of the method. i.e.

    if (!row.deleteModel) {
        return row.rowActions;
    }

Reply:
agreed.